xarray/tests/test_indexing.py

```diff
@@ -622,3 +622,8 @@ def test_create_mask_error():
 def test_posify_mask_subindexer(indices, expected):
     actual = indexing._posify_mask_subindexer(indices)
     np.testing.assert_array_equal(expected, actual)
+
+
+def test_indexing_1d_object_array():
+    testarray = DataArray(np.array((np.arange(3), np.arange(6))))
+    testarray[0]
```

Comment:
Well...
I guess you mean:
expected_array = np.arange(3)
expected = DataArray(expected_array)
assert_identical(expected, testarray[0])

Reply:
`expected_array = np.arange(3)` gives you an array with shape `(3,)` and int dtype. You want a scalar NumPy with shape `()` and object dtype.

Reply:
OK, I get you point :-)
Main issue I still have then is that I'm not able to get the 'np.arange(3) array'
from the 'expected'... 
because 'expected.data' is king of empty :-(
expected.data.shape == ()  therefore expected.data[0] raise an IndexError
How would you access the content of 'expected' ?